Platon/backend/app/search_engine/views.py

```diff
@@ -308,7 +327,77 @@ def get(self):
         else:
             return make_response(jsonify({"error": "Missing data fields or invalid data."}), 400)
 
+@search_engine_ns.route("/upcoming_events")
+class UpcomingEventsSearchAPI(Resource):
 
+    @api.doc(responses={401: 'Account Problems', 400: 'Input Format Error', 500: 'Database Connection Error'})
+    @api.response(200, 'Valid Search Result', upcoming_event_list_model)
+    @api.expect(upcoming_events_search_parser)
+    def get(self):
+        form = UpcomingEventsSearchForm(request.args)
+        if form.validate():
+            search_query = form.search_query.data.lower()
+            # Remove the punctuation in the search query
+            search_query = SearchEngine.remove_punctuation(search_query)
+            # Tokenize the search query
+            tokens = search_query.split()
+            # Get stopwords form the API
+            stopwords = SearchEngine.get_stopwords()
+            # Remove stopwords from the list of tokens
+            tokens = list(set(tokens) - set(stopwords))
+            # Get the semantically related list for our tokens
+            tokens = SearchEngine.semantic_related_list(tokens)
+            # List of user dictionaries that will be send as a search output
+            result_list = []
+            # Score of each user record
+            result_id_score = []
+            for token, score in tokens:
+                try:
+                    query = '(LOWER(id) REGEXP ".*{0}.*" OR LOWER(title) REGEXP ".*{0}.*" \
+                    OR LOWER(acronym) REGEXP ".*{0}.*" OR LOWER(location) REGEXP ".*{0}.*") \
+                    OR LOWER(date) REGEXP ".*{0}.*" OR LOWER(deadline) REGEXP ".*{0}.*" OR LOWER(link) REGEXP ".*{0}.*"'  \
```

Comment:
Why do you search for the ID of the event? Actually, I think ID has no meaning for the users that will use this endpoint. If you have a valid reason please state your reason.

Reply:
I think the link also has no meaning to the user.

Reply:
I removed them.